pkg/commands/alerts/list.go

```diff
@@ -0,0 +1,115 @@
+package alerts
+
+import (
+	"errors"
+	"io"
+
+	"github.com/fastly/go-fastly/v9/fastly"
+
+	"github.com/fastly/cli/pkg/argparser"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+)
+
+// NewListCommand returns a usable command registered under the parent.
+func NewListCommand(parent argparser.Registerer, g *global.Data) *ListCommand {
+	c := ListCommand{
+		Base: argparser.Base{
+			Globals: g,
+		},
+	}
+	c.CmdClause = parent.Command("list", "List Alerts")
+
+	// Optional.
+	c.RegisterFlagBool(c.JSONFlag()) // --json
+	c.CmdClause.Flag("cursor", "Pagination cursor (Use 'next_cursor' value from list output)").Action(c.cursor.Set).StringVar(&c.cursor.Value)
+	c.CmdClause.Flag("limit", "Maximum number of items to list").Action(c.limit.Set).IntVar(&c.limit.Value)
+	c.CmdClause.Flag("name", "Name of the definition").Action(c.definitionName.Set).StringVar(&c.definitionName.Value)
+	c.CmdClause.Flag("sort", "Sort by one of the following [name, created_at, updated_at]").Action(c.sort.Set).StringVar(&c.sort.Value)
+	c.CmdClause.Flag("order", "Sort by one of the following [asc, desc]").Action(c.order.Set).StringVar(&c.order.Value)
+	c.CmdClause.Flag(argparser.FlagServiceIDName, "ServiceID of the definition").Action(c.serviceID.Set).StringVar(&c.serviceID.Value)
+
+	return &c
+}
+
+// ListCommand calls the Fastly API to list appropriate resources.
+type ListCommand struct {
+	argparser.Base
+	argparser.JSONOutput
+
+	cursor         argparser.OptionalString
+	limit          argparser.OptionalInt
+	definitionName argparser.OptionalString
+	serviceID      argparser.OptionalString
+	sort           argparser.OptionalString
+	order          argparser.OptionalString
+}
+
+// Exec invokes the application logic for the command.
+func (c *ListCommand) Exec(_ io.Reader, out io.Writer) error {
+	if c.Globals.Verbose() && c.JSONOutput.Enabled {
+		return fsterr.ErrInvalidVerboseJSONCombo
+	}
+
+	input, err := c.constructInput()
+	if err != nil {
+		return err
+	}
+
+	definitions, err := c.Globals.APIClient.ListAlertDefinitions(input)
+	if err != nil {
+		return err
+	}
+
+	if ok, err := c.WriteJSON(out, definitions); ok {
+		return err
+	}
+
+	definitionsPtr := make([]*fastly.AlertDefinition, len(definitions.Data))
+	for i := range definitions.Data {
+		definitionsPtr[i] = &definitions.Data[i]
+	}
+
+	if c.Globals.Verbose() {
+		printVerbose(out, &definitions.Meta, definitionsPtr)
+	} else {
+		printSummary(out, &definitions.Meta, definitionsPtr)
+	}
```

Comment:
I'm not sure how useful it is to display the cursor information.

A CLI user is going to expect the list command to return all data and not stop after one page of paginated data. 

So we should either automatically follow all cursors until we've read all the data or if that could be a memory concern, we should prompt the user to "request another page of data". You can see an example of this approach here...

https://github.com/fastly/cli/blob/6c1ec982475e698bd347c94622f857d9e4f44c90/pkg/commands/secretstoreentry/list.go#L63-L75

Reply:
Sounds good, we have some hard limits in the API that we cannot go over so I will add this suggested snippet.